### PR TITLE
peppin

### PR DESCRIPTION
**Pull request by jshawl**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+- Set up public key ssh authentication
+- Deploy a nodejs app
+- Configure a reverse proxy with nginx
+
+## Set up droplet
+
+Please use my referral code - well both get 10 bucks in credits!
+
+<http://jshawl.com/do>
+
```
Comment (nolds9) [inline readme.md:23]: After you signup, click "Create a Droplet"...

File: readme.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+All the defaults are fine. Choose the cheapest droplet.
+
+All the way at the bottom, add your ssh key.
+
+You can get your ssh key by typing in
+
+```
+$ cat ~/.ssh/id_rsa.pub
+```
```
Comment (nolds9) [inline readme.md]: what if they don't have an ssh key already generated?

File: readme.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+This will allow you to log into the server without having to type your password.
+
+```
+$ ssh root@i.p.address.here
+```
```
Comment (nolds9) [inline readme.md]: Could turn this into a CFU - what does it mean to "ssh into a server''? What can we do?

File: readme.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+## Create a user and password
+
+```
+$ useradd -m theusernameyouwant -G sudo -s /bin/bash
+$ passwd theusernameyouchose
+```
+
+## `/etc/hosts`
+
+Remembering an IP address is difficult and time consuming. Let's fake a domain name
```
Comment (nolds9) [inline readme.md]: maybe a little recap about how the IP address will be used in the scheme of things

File: readme.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+by editing our hosts file. This file is consulted before asking DNS servers what
+ip address a domain points to.
+
+```
+sudo nano /etc/hosts
```
Comment (nolds9) [inline readme.md]: will students be familiar with nano commands?

File: readme.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+add the following line at the bottom:
+
+```
+your.ip.address.here example.com
+```
+
+Feel free to replace example.com with whatever domain you'd like.
+
+Test it out!
```
Comment (nolds9) [inline readme.md]: how?

File: readme.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+$ ssh yourusername@example.com
+$ mkdir .ssh/
+```
+
+## Pub key auth
+
+Typing in a password is difficult and time consuming. We can prevent having to type
+in a password by configuring public key authentication.
+
+Locally...
```
Comment (nolds9) [inline readme.md]: I get this concept, but some students might be confused by what you mean by locally....

File: readme.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+```
+
+You'll know you did this correctly if you are logged in, but not prompted for a password.
+
+## Install git
+
+```
+$ sudo apt-get install git
+$ mkdir ~/git
+$ git init --bare ~/git/yourdomain.com
```
Comment (amaseda) [inline readme.md]: What does `--bare` mean?

File: readme.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+cd to your local copy of your site
+
+```
+$ git remote add production yourusername@yourdomain.com:git/yourdomain.com
+$ git push production master
+```
+
+## Install nginx
+
```
Comment (amaseda) [inline readme.md]: Add note about what nginx is

**Review by nolds9 — COMMENTED** [review]
Looks great so far, overall think it might be helpful to add the why for the setup

File: readme.md [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+## post update hook
+
+You can currently deploy your site by pushing to your git repo, and pulling into
+the website's directory, but git can do this automatically for you.
+
+```
+$ cd ~/git/yourdomain.com/hooks
+$ mv post-update.example post-update
+$ chmod +x post-update
```
Comment (amaseda) [inline readme.md]: What does `+x` mean?

**Pull request merged**